java/server/src/org/openqa/grid/internal/BaseRemoteProxy.java

```diff
@@ -342,7 +309,7 @@ public TestSession getNewSession(Map<String, Object> requestedCapability) {
       return null;
     }
     // any slot left for the given app ?
-    for (TestSlot testslot : testSlots) {
+    for (TestSlot testslot : getTestSlots()) {
```

Comment:
getTotalUsed(), hasCapability() -- other places in this class still enumerate on the private immutable List `testSlots`. Will that be a problem per you intended change?